packages/vm/src/imports.rs

```diff
@@ -70,22 +63,16 @@ mod errors {
 
     /// canonicalize_address errors (-1_002_0xx)
     pub mod canonicalize {
-        /// An unknown error when canonicalizing address
-        pub static UNKNOWN: i32 = -1_002_000;
         /// The input address (human address) was invalid
         pub static INVALID_INPUT: i32 = -1_002_001;
     }
 
-    /// humanize_address errors (-1_002_1xx)
-    pub mod humanize {
-        /// An unknonw error when humanizing address
-        pub static UNKNOWN: i32 = -1_002_100;
-    }
+    // /// humanize_address errors (-1_002_1xx)
+    // pub mod humanize {
+    // }
 
     /// query_chain errors (-1_003_0xx)
     pub mod query_chain {
-        /// An unknown error in query_chain
-        // pub static UNKNOWN: i32 = -1_003_000;
```

Comment:
The idea of those unused errors is to make clear why the numbers are the ones we use. The numbering scheme needs to stabilize no matter of we use them right now or not. The xxxxx0 of a block is always the fallback error. To make it easy to understand, they are commentes out.

I'm not saying the handling of those numbers is great at the moment. It was worse before and improved. But we need to get to a point were we have a proper namespacing system that makes sense, such that the contract can rely on some of those numbers no matter how the VM changes.

Reply:
This all makes sense, but if we hit a situation where we don't know exactly what's wrong, we shouldn't be returning "I don't know what's going on" to the contract, but it should cause a Trap by returning `Err` from the methods below, and halting the contract (as i've done). The contract should only ever observe errors that were directly caused by its own actions, and that it has a reasonable way to handle. It is worth checking whether all cases where i throw a trap to wasmer today are indeed cases that can be reported back to the contract.

Reply:
So, should I renumber the errors?

Reply:
Let's keep them unchanged for now. Thanks for all the thoughts you put into that.

We can look into that again when working on _// TODO convert these numbers to a single enum._ for 0.9. Since the numbers are only used inside the standard library, the effect on the contract developer is not too big.

Reply:
Not quite - changing them is a breaking change for the contract ABI. I think it's just an hour or two of work to define an enum similar to `GoResult` in `go-cosmwasm` which can later be matched against in updates to `cosmwasm-std`

Reply:
> changing them is a breaking change for the contract ABI

Yes I know. But if the user updates both standard library and VM from 0.8 to 0.9, they will not even notice the change.

Reply:
Feel free to open a ticket or PR. Resolving this for now here.